src/WindowsUtils/BusyLoopUtils.cpp

```diff
@@ -0,0 +1,109 @@
+// Copyright (c) 2022 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "WindowsUtils/BusyLoopUtils.h"
+
+#include <absl/base/casts.h>
+#include <absl/strings/str_format.h>
+
+#include "OrbitBase/GetLastError.h"
+#include "WindowsUtils/ReadProcessMemory.h"
+#include "WindowsUtils/SafeHandle.h"
+#include "WindowsUtils/WriteProcessMemory.h"
+
+namespace orbit_windows_utils {
+
+namespace {
+
+ErrorMessage GetLastErrorForFunction(std::string_view function) {
+  return ErrorMessage(
+      absl::StrFormat("Calling \"%s\": %s", function, orbit_base::GetLastErrorAsString()));
+}
+
+ErrorMessageOr<void> FlushInstructionCache(HANDLE process_handle, void* address, size_t size) {
+  if (::FlushInstructionCache(process_handle, address, size) == 0) {
+    return GetLastErrorForFunction("FlushInstructionCache");
+  }
+  return outcome::success();
+}
+
+ErrorMessageOr<void> WriteInstructionsAtAddress(const std::string& byte_buffer, void* address) {
```

Comment:
Why is the byte buffer a string? That feels wrong.

Reply:
It's initially because of protobufs using std::string as byte buffer and our Read/WriteProcessMemory adopting std::string to easily move to a protobuf. Added a comment. See b/228234988. I'll change it in another PR.